lib/streamlit/commands/execution_control.py

```diff
@@ -87,3 +90,51 @@ def experimental_rerun() -> NoReturn:
     # be seen.
     _LOGGER.warning(msg)
     rerun()
+
+
+@gather_metrics("switch_page")
+def switch_page(page: str) -> NoReturn:  # type: ignore[misc]
+    """Switch the current programmatically page in a multi-page app.
+    When `st.switch_page()` is called with a page, the current page script is halted
+    and the requested page script will be queued to run from the top.
+    Parameters
+    ----------
+    page: str
+        The file path, relative to the main script, of the page to switch to.
+    """
+
+    ctx = get_script_run_ctx()
+
+    ctx_main_script = ""
+    if ctx:
+        ctx_main_script = ctx.main_script_path
+
+    main_script_path = os.path.join(os.getcwd(), ctx_main_script)
```

Comment:
What happens if ctx is not available, we just ignore?

Seems like we should be just return and log an warning that switch_page does not work because there's no context?